contracts/recovery-contracts/WalletSimple.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity ^0.4.18;
+import "./Forwarder.sol";
+import "./CloneFactory.sol";
+import "../ERC20Interface.sol";
+/**
+ *
+ * WalletSimple
+ * ============
+ *
+ * Basic singleSig wallet designed to recover funds.
+ *
+ */
+contract WalletSimple is CloneFactory {
+
+  // Public fields
+  address public signer;
+  address public implementationAddress;
+
+  function WalletSimple (address _signer, address _implementationAddress) public {
+
+    signer = _signer;
+    implementationAddress = _implementationAddress;
+
+  }
+
+  /**
+   * Modifier that will execute internal code block only if the sender is an authorized signer on this wallet
+   */
+  modifier onlySigner {
+    if (signer != msg.sender) {
+      revert();
+    }
+    _;
+  }
+
+ /**
+   * Gets called when a transaction is received without calling a method
+   */
+  function() public payable {
+  }
+
+  /**
+   * Create a new contract that forwards funds to this contract
+   */
+  function createForwarder() external {
+    address clone = createClone(implementationAddress);
+    Forwarder(clone).init(address(this));
```

Comment:
is this sufficient for creating a forwarder with the same deterministic address we would expect as with our regular contracts? 

I would assume we'd want to be using the nonce value of the tx to derive this address? or i suppose this is happening implicitly in the EVM?

Reply:
Forwarder is calculated using value of wallet address and its nonce(wallet address), same as old contracts